File: playground/csharp/Playground/Playgrounds/AgentStudio.cs
using Algolia.Search.Clients;
using Algolia.Utils;
using Microsoft.Extensions.Logging;

namespace Algolia.Playgrounds;

public class AgentStudioPlayground : IPlayground
{
  private readonly AgentStudioClient _client;

  public AgentStudioPlayground(Configuration configuration)
  {
    var loggerFactory = LoggerFactory.Create(i => i.AddFilter("Algolia", LogLevel.Information)
      .AddConsole());
    var config = new AgentStudioConfig(configuration.AppId, configuration.AdminApiKey, "us");
    _client = new AgentStudioClient(config, loggerFactory);
  }

  public async Task Run()
  {
    PlaygroundHelper.Hello("Starting Agent Studio API playground");
    try
    {
      var response = await _client.ListAgentsAsync().ConfigureAwait(false);
      Console.WriteLine("List of agents:");
      foreach (var agent in response.Data)
      {
        Console.WriteLine($"- {agent.Name} (ID: {agent.Id})");
      }
    }
    catch (Exception e)
    {
      Console.WriteLine($"Error: {e.Message}");
    }
  }
}

File: playground/javascript/node/agentStudio.ts
import { ApiError } from '@algolia/client-common';
import { agentStudioClient } from '@algolia/agent-studio';

const appId = process.env.ALGOLIA_APPLICATION_ID || '**** APP_ID *****';
const apiKey = process.env.ALGOLIA_ADMIN_KEY || '**** ADMIN_KEY *****';

// Init client with appId and apiKey
const client = agentStudioClient(appId, apiKey);

async function testAgentStudio() {
  try {
    const response = await client.listAgents();

    console.log('List of agents:');
    for (const agent of response.data) {
      console.log(`- ${agent.name} (ID: ${agent.id})`);
    }
  } catch (e) {
    if (e instanceof ApiError) {
      return console.log(`[${e.status}] ${e.message}`, e.stackTrace);
    }

    console.log('[ERROR]', e);
  }
}

testAgentStudio();
